Microsoft.Toolkit.Mvvm/ComponentModel/ObservableValidator.cs

```diff
@@ -0,0 +1,159 @@
+        protected bool ValidateProperty(object? value, [CallerMemberName] string? propertyName = null)
+        {
+            if (propertyName is null)
+            {
+                ThrowArgumentNullExceptionForNullPropertyName();
+            }
+
+            bool errorsChanged = false;
+
+            // Clear the errors for the specified property, if any
+            if (this.errors.TryGetValue(propertyName!, out List<ValidationResult>? propertyErrors) &&
+                propertyErrors.Count > 0)
+            {
+                propertyErrors.Clear();
+
+                errorsChanged = true;
+            }
+
+            List<ValidationResult> results = new List<ValidationResult>();
+
+            // Validate the property
+            bool isValid = Validator.TryValidateProperty(
+                value,
+                new ValidationContext(this, null, null) { MemberName = propertyName },
+                results);
+
+            // Update the state and/or the errors for the property
+            if (isValid)
+            {
+                if (errorsChanged)
+                {
+                    this.totalErrors--;
+                }
+            }
+            else
+            {
+                if (!errorsChanged)
+                {
+                    this.totalErrors++;
+                }
+
+                // We can avoid the repeated dictionary lookup if we just check the result of the previous
+                // one here. If the retrieved errors collection is null, we can log the ones produced by
+                // the validation we just performed. We also don't need to create a new list and add them
+                // to that, as we can directly set the resulting list as value in the mapping. If the
+                // property already had some other logged errors instead, we can add the new ones as a range.
+                if (propertyErrors is null)
+                {
+                    this.errors.Add(propertyName!, results);
+                }
+                else
+                {
+                    propertyErrors.AddRange(results);
+                }
+
+                errorsChanged = true;
+            }
+
+            // Only raise the event once if needed. This happens either when the target property
+            // had existing errors and is now valid, or if the validation has failed and there are
+            // new errors to broadcast, regardless of the previous validation state for the property.
+            if (errorsChanged)
+            {
+                ErrorsChanged?.Invoke(this, new DataErrorsChangedEventArgs(propertyName));
+            }
+
+            return isValid;
+        }
+
+        /// <summary>
+        /// Throws an <see cref="ArgumentNullException"/> when a property name given as input is <see langword="null"/>.
+        /// </summary>
+        private static void ThrowArgumentNullExceptionForNullPropertyName()
```

Comment:
Is this still used?

Reply:
Yup, I'm using this to validate the input `propertyName` arg for the `ValidateProperty` method:

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/77ed6e8bfdd9e129d2dfb350cbe8e0f673be0c08/Microsoft.Toolkit.Mvvm/ComponentModel/ObservableValidator.cs#L84-L87